tests/test_docker_cat.py

```diff
@@ -0,0 +1,539 @@
+        """
+        sensors = (
+            "Sensor C++ (Community) SquidSensor [cxx]",
+            "Sensor SonarFrama-C [framac]"
+        )
+        self.language("C/C++", "c++", "c_cpp", sensors, "c-dummy-project", 0, "CNES_C_A", 1)
+        # 0 issue are expected with the Sonar way Quality Profile for
+        # C++ (Community) because it does not have any rule enabled.
+    
+    def test_language_fortran_77(self):
+        """As a user of this image, I want to analyze a fortran 77 project
+        so that I can see its level of quality on the SonarQube server.
+        """
+        sensors = (
+            "INFO: Sensor Sonar i-Code [icode]",
+        )
+        self.language("Fortran 77", "f77", "fortran77", sensors, "fortran77-dummy-project", 11)
+
+    def test_language_fortran_90(self):
+        """As a user of this image, I want to analyze a fortran 90 project
+        so that I can see its level of quality on the SonarQube server.
+        """
+        sensors = (
+            "INFO: Sensor Sonar i-Code [icode]",
+        )
+        self.language("Fortran 90", "f90", "fortran90", sensors, "fortran90-dummy-project", 14)
+
+    def test_language_java(self):
+        """As a user of this image, I want to analyze a java project
+        so that I can see its level of quality on the SonarQube server.
+        """
+        sensors = (
+            "INFO: Sensor CheckstyleSensor [checkstyle]",
+            "INFO: Sensor FindBugs Sensor [findbugs]",
+            "INFO: Sensor PmdSensor [pmd]",
+            "INFO: Sensor CoberturaSensor [cobertura]"
+        )
+        self.language("Java", "java", "java", sensors, "java-dummy-project", 3, "CNES_JAVA_A", 6)
+
+    def test_language_python(self):
+        """As a user of this image, I want to analyze a Python project
+        so that I can see its level of quality on the SonarQube server.
+        """
+        self.language("Python", "py", "python", (), "python-dummy-project", 2, "CNES_PYTHON_A", 3)
+
+    def test_language_shell(self):
+        """As a user of this image, I want to analyze a shell project
+        so that I can see its level of quality on the SonarQube server.
+        """
+        sensors = (
+            "INFO: Sensor Sonar i-Code [icode]",
+        )
+        self.language("Shell", "shell", "shell", sensors, "shell-dummy-project", 58)
+
+    # Test analysis tools
+    def test_tool_cppcheck(self):
+        """As a user of this image, I want to run cppcheck from within a container
+        so that it produces a report.
+        """
+        ref = "tests/c_cpp/reference-cppcheck-results.xml"
+        output = "tests/c_cpp/tmp-cppcheck-results.xml"
+        cmd = f"cppcheck --xml-version=2 tests/c_cpp/cppcheck/main.c --output-file={output}"
+        self.analysis_tool("cppcheck", cmd, ref, output, False)
+
+    def test_tool_frama_c(self):
+        """As a user of this image, I want to run Frama-C from within a container
```

Comment:
Codacy found an issue: [1 blank line required between summary line and description (found 0) (D205)](https://app.codacy.com/gh/cnescatlab/docker-cat/pullRequest?prid=6253839)